app/controllers/reports_controller.rb

```diff
@@ -10,6 +10,6 @@ def index
       body_part_id: params[:body_part_id]
     }
 
-    @report = Report.new(filter_params)
+    @report ||= Report.build_report(filter_params)
   end
```

Comment:
Rubocop says memoizing an instance variable is not cool. Additionally, it should be named `@index`. What gives?

Reply:
Rubocop expects memoized objects to be created inside a method with the same name, but Im unsure why it doesn't want you to use it here. Can you post the cop?

Reply:
@mmdotz 
```
app/controllers/reports_controller.rb:13:5: C: Naming/MemoizedInstanceVariableName: Memoized variable @report does not match method
 name index. Use @index instead.
    @report ||= Report.build_report(filter_params)
    ^^^^^^^
```

Reply:
I don't think you'd need memoization here since we're in the controller. The controller is setting `@report` and then handing it off to the view to render, so the instance variable will only be set once and then referenced however many times your view needs it.

Reply:
Ah, okay. I was thinking for a moment that every time `@report` is called in the view that it would run the `build_report` method and that would be costly.